drivers/blueview_driver/src/BlueViewRosDriver.cpp

```diff
@@ -0,0 +1,128 @@
+	nh.param<bool>("color/enable", do_color, false);
+	nh.param<bool>("raw/enable", do_raw, true);
+	if (do_grayscale) {
+		grayscale_img.reset(new cv_bridge::CvImage());
+		grayscale_img->encoding = "mono16";
+        grayscale_img->header.frame_id = frame_id;
+		grayscale_pub = image_transport.advertise("image_mono", 1);
+	}	
+	if (do_color) {
+
+        // Get and load color map
+        std::string color_map_file;
+        if (nh.getParam("color/map_file", color_map_file))
+            sonar.loadColorMapper(color_map_file);
+
+		color_img.reset(new cv_bridge::CvImage());
+		color_img->encoding = "bgra8";
+        grayscale_img->header.frame_id = frame_id;
+		color_pub = image_transport.advertise("image_color", 1);
+	}
+	if (do_raw) {
+		ping_msg.reset(new sonar::BlueViewPing);
+		raw_pub = nh.advertise<sonar::BlueViewPing>("ranges", 5);
+	}
+
+	std::string params;
+	if (nh.getParam("file", params))
+		sonar.init(BlueViewSonar::ConnectionType::FILE, params);
+	else if (nh.getParam("device", params))
+		sonar.init(BlueViewSonar::ConnectionType::DEVICE, params);
+	else throw std::runtime_error("Can not connect: neither 'file' or 'device' param set");
+
+	// Set Ranges, meters
+	BVTSDK::Head& head  = sonar.getHead();
+	float range_lower, range_upper;
+	if (nh.getParam("range/start", range_lower)) head.SetStartRange(range_lower);
+	if (nh.getParam("range/stop",  range_upper)) head.SetStopRange(range_upper);
+
+	// Handle fluid type, enum string
+	std::string fluid_type;
+	if (nh.getParam("fluid_type", fluid_type)) {
+		if (fluid_type == "saltwater") head.SetFluidType(BVTSDK::FluidType::Saltwater);
+		else if (fluid_type == "freshwater") head.SetFluidType(BVTSDK::FluidType::Freshwater);
+		else if (fluid_type == "other") head.SetFluidType(BVTSDK::FluidType::Other);
+		else ROS_ERROR("Fluid type (%s) invalid, must be 'saltwater', 'freshwater', or 'other'", fluid_type.c_str());
+	}
+
+	// Set sound speed, m/s
+	int sound_speed;
+	if (nh.getParam("sound_speed", sound_speed)) head.SetSoundSpeed(sound_speed);
+	
+	// Set analog gain adjustment in dB
+	float gain;
+	if (nh.getParam("gain_adjustment", gain)) head.SetGainAdjustment(gain);
+	
+	// Set "time variable analog gain", in dB/meter,
+	float tvg;
+	if (nh.getParam("tvg_slope", tvg)) head.SetTVGSlope(tvg);
+	
+	// Set dynamic power managment
+	bool dynamic_power;
+	if (nh.getParam("dynamic_power_management", dynamic_power)) head.SetDynamicPowerManagement(dynamic_power);
+	
+	// Set ping interval
+	float ping_interval;
+	if (nh.getParam("ping_interval", ping_interval)) head.SetPingInterval(ping_interval);
+
+	// Start loop
+	double period_seconds;
+	nh.param<double>("period_seconds", period_seconds, 0.1);
+	timer = nh.createTimer(ros::Duration(period_seconds),std::bind(&BlueViewRosDriver::loop, this, std::placeholders::_1));
+}
+void BlueViewRosDriver::loop(const ros::TimerEvent&)
+{
+	sonar.getNextPing();
+
+	// Do images
+	if(do_grayscale || do_color)
+	{
```

Comment:
As per the [ros style guide](http://wiki.ros.org/CppStyleGuide#Formattingl), braces should all be on their own line. Can you apply this everywhere else in the PR?

Also, all the tabs should be converted to spaces.

I'd recommend reading through the style guide since this is another package that we could potentially contribute.